GetIntoTeachingApi/Models/TeachingEvent.cs

```diff
@@ -116,11 +116,14 @@ protected override void FinaliseEntity(Entity source, ICrmService crm, Organizat
         {
             var existingEvent = crm.GetTeachingEvent(ReadableId);
 
-            bool removeBuilding = Building == null && existingEvent.Building != null;
-
-            if (removeBuilding)
+            if (existingEvent != null)

```

Comment:
Don't feel strongly but you could drop the nested `if`s by using a safe navigation here:

```
bool removeBuilding = Building == null && existingEvent?.Building != null;
```